### PR TITLE
Support for remote validation

### LINKED ISSUE #1: Remote validation

### PR DESCRIPTION
**Pull request by AleksandarDev**
- Added support for Promise validation functinons
- Added cypress React component tests for useValidation, submitForm and validateFields

Closes #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +35,16 @@
...
         "@babel/core": "^7.10.1",
-        "@babel/preset-env": "^7.10.1"
+        "@babel/preset-env": "^7.10.1",
+        "@babel/preset-react": "^7.12.1",
+        "@cypress/code-coverage": "^3.8.2",
+        "babel-loader": "^8.1.0",
+        "babel-plugin-istanbul": "^6.0.0",
+        "cypress": "^5.5.0",
+        "cypress-react-unit-test": "^4.16.7",
+        "react": "^16.13.1",
```
Comment (radovix) [inline package.json:45]: @AleksandarDev should we have `react` and `react-dom` as `devDependencies` or as `peerDependencies`?

File: cypress/testComponents/EmailComponentWithFields.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react';
+import { validateFields } from '../../src/formUtils';
+import { useValidation } from '../../src/useValidation';
+
+const EmailComponentWithFields = (props) => {
+    const email = useValidation(props.email, props.emailValidation);
+
+    useEffect(() => {
+        var validationResult = validateFields([email]);
```
Comment (radovix) [inline cypress/testComponents/EmailComponentWithFields.jsx:9]: `validateFields` should receive object, not array. Right way to write that line should be `constvalidationResult = validateFields({ email });`.

Comment (AleksandarDev) [reply cypress/testComponents/EmailComponentWithFields.jsx:9]: Readme needs a fix too.

File: package.json (hunk 1 above)
Comment (AleksandarDev) [reply package.json:45]: dev, used only for running tests

Comment (radovix) [reply package.json:45]: Hooks rely on it as well.

File: cypress/testComponents/EmailComponentWithFields.jsx (hunk 2 above)
Comment (radovix) [reply cypress/testComponents/EmailComponentWithFields.jsx:9]: 👍

File: package.json (hunk 1 above)
Comment (AleksandarDev) [reply package.json:45]: Hooks will use react installed as peerdep

Comment (radovix) [reply package.json:45]: Makes sense.

File: cypress/testComponents/EmailComponentWithFields.jsx (hunk 2 above)
Comment (radovix) [reply cypress/testComponents/EmailComponentWithFields.jsx:9]: @AleksandarDev I will merge the PR when you change this. You can suggest how the documentation can be modified as well (no need to change it now tho).

Comment (AleksandarDev) [reply cypress/testComponents/EmailComponentWithFields.jsx]: @radovix Fixed and created new issue for documentation

**Review by radovix — APPROVED** [review]

**Pull request merged**